src/rev_ai/apiclient.py

```diff
@@ -324,6 +325,18 @@ def get_account(self):
         """
         url = urljoin(self.base_url, 'account')
         response = self.session.get(url)
-        response.raise_for_status()
+        self._handle_api_failure(response)
```

Comment:
An alternate pattern you could have used:
```
@rev_http_wrapper
def get_account(self):
    url = urljoin(self.base_url, 'account')
    response = self.session.get(url)
    response.raise_for_status()

    return Account.from_json(response.json())
```

where you have rev_http_wrapper do your thing with the exception handling.

Why would I choose this over the inline changes? The decorator provides some abstraction -- this keeps the method body simpler with familiar raise_for_status() calls and tells readers not to worry about the http handling behavior. After all, we are more concerned with capturing the HTTPError than the actual behavior of `raise_for_status`. Also, if you want to modify the http error handling behavior in the future, you have a bit more power with a decorator than with the `self._handle_api_failure(response)` method call. You should weigh this against your reasons to use a private method to wrap the `raise_for_status` call.

Reply:
This decorator abstraction is definitely nice, I'll have to look more into decorators in python. I have future plans to completely abstract the response library and error handling from this client in the future (similar to what we did for the node sdk if you've seen that). 

Maybe it makes sense to just do that in this issue